Payments introduction, first attempt (#116)
SwedbankPay/developer.swedbankpay.com: payments/index.md

{% include alert-development-section.md %}

## Introduction

Payments is our off-the-rack assortment of payment instruments. You can pick and
choose among them to tailor the instrument package best suited for your
business. Is Vipps all you need? Go for Vipps only. Do you want Vipps and
invoice? Implement Vipps and invoice. Want them all? Have them all. Each payment
instrument is set up with a separate contract and integration. 

With a couple of exceptions, our payment instruments are available on three
platforms. Choose between our easy-to-use PCI compliant platforms Redirect

> and Seamless View - or use Swedbank Pay Direct API to integrate directly. Our

Reviewer: ```suggestion
and Seamless View – or use Swedbank Pay Direct API to integrate directly. Our
```

payment instruments and their platform availability are listed in the table
below.

{:.table .table-striped}
|                                | Payment instrument                  | Seamless View | Redirect | Region                                                                               |
| :----------------------------: | :---------------------------------- | :-----------: | :------: | :----------------------------------------------------------------------------------- |
|     ![CardIcon][card-icon]     | [Card][card]                 |    ✔︎︎︎︎︎     |  ✔︎︎︎︎︎  | ![EarthIcon][earth-icon]                                                             |

> |  ![InvoiceIcon][invoice-icon]  | [Swedbank Pay Invoice][invoice]     |    ✔︎︎︎︎︎     |  ✔︎︎︎︎︎  | ![nor][nor-flag] ![swe][swe-flag] ![fin][fin-flag]                                   |

Reviewer: ```suggestion
|  ![InvoiceIcon][invoice-icon]  | [Invoice][invoice]     |    ✔︎︎︎︎︎     |  ✔︎︎︎︎︎  | ![nor][nor-flag] ![swe][swe-flag] ![fin][fin-flag]                                   |
```

> | ![EnvelopeIcon][envelope-icon] | [Swedbank Pay Web Invoice][invoice] |    ✔︎︎︎︎︎     |  ✔︎︎︎︎︎  | ![nor][nor-flag] ![swe][swe-flag]                                                    |

Reviewer: ```suggestion
| ![EnvelopeIcon][envelope-icon] | [Web Invoice][invoice] |    ✔︎︎︎︎︎     |  ✔︎︎︎︎︎  | ![nor][nor-flag] ![swe][swe-flag]                                                    |
```

|  ![MobilePay][mobilepay-logo]  | [MobilePay][mobile-pay]            |               |  ✔︎︎︎︎︎  | ![dan][dan-flag] ![fin][fin-flag]                                                                      |

## Prerequisites

To start integrating Swedbank Pay Payments, you need the following:

* An [HTTPS][https] enabled web server.
* An agreement which includes Swedbank Pay Payments.
* Credentials (Merchant Access Token) from Swedbank Pay retrieved from
  Swedbank Pay Admin.

## Platform options

Here are our three platform options at a glance. You can read more about the
integration process by visiting the sections for each payment instrument.

### Seamless View

With [Seamless Views][seamless-view] you can initiate the payment process
directly in an `iframe` on your site. If you prefer that the consumer isn't
redirected away from you, this option enables you to embed our payment page
seamlessly into your shopping experience.

### Redirect

The [Redirect][redirect] platform redirects the consumers to a Swedbank Pay
hosted payment page. The consumer will be redirected back to your page when the
payment is completed.

### Direct

> [Direct][direct] is the option where you integrate directly using our Direct

Reviewer: ```suggestion
Direct is the option where you integrate directly using our Direct
```

API. This is an integration with the most flexibility and opportunities. If you

> want to offer card payments and choose this option, you have to be PCI-DSS

Reviewer: ```suggestion
want to offer Card Payments and choose this option, you have to be PCI-DSS
```

                      for all payment instruments, reducing complexity and
                      enabling a more straightforward integration." %}

> There are to main payment types, *two-phase* and *one-phase* payments. The two

Reviewer: ```suggestion
There are two main payment types, *two-phase* and *one-phase* payments. The two
```

seem very similar from a consumer's point of view, but there are key differences
you should know about.

*Two-phase payments* ([Card][card], [Invoice][invoice],
[MobilePay][invoice], [Vipps][vipps])

A two-phase payment is performed in two steps – an `authorization` which
reserves the consumer's funds, and a `capture` of the funds at a later time,
usually when the goods are shipped.

> This is the most common payment type, and it is used by card payments, Vipps

Reviewer: ```suggestion
This is the most common payment type, and it is used by Card Payments, Vipps
```

> payments, MobilePay payments, invoice payments and credit account payments. A

Reviewer: ```suggestion
Payments, MobilePay payments, Invoice Payments and Credit Account Payments. A
```

capture of an invoice will *not* capture any funds, but trigger the invoice
distribution and send it to the consumer.

*One-phase payments* ([Direct Debit][direct-debit], [Swish][swish],
[Card][card])

There are two types of one-phase payments – `sale` and `autoCapture`.

`sale` is used by payment instruments such as Direct Debit and [Swish][swish].
These payments will have a `sale` transaction instead of the `authorization` and
`capture`. The funds will be captured from the consumer straight away.

> `autoCapture` is only available for card payments. The mechanics work the same

Reviewer: ```suggestion
`autoCapture` is only available for Card Payments. The mechanics work the same
```

way as a two-phase payment, with two separate transactions – one for the
`authorization` and one for the `capture`. 

As the name implies, the capture transaction is performed automatically when the
authorization is successful. Because of this, `autoCapture` should only be used
when dealing with digital products, since they are shipped instantly.

As the funds are captured instantly, `cancel` is not available for either of the
one-phase payments. `abort` and `reversal` can be performed the same way as with
two-phase payments.

## The Payment Object

>   The payment is the container object that holds all transactions

Reviewer: ```suggestion
The payment is the container object that holds all transactions
```

>   created during the payment process. When Swedbank Pay receives the payment

Reviewer: ```suggestion
created during the payment process. When Swedbank Pay receives the payment
```

>   request body (in JSON format), a payment is created and you will be given a

Reviewer: ```suggestion
request body (in JSON format), a payment is created and you will be given a
```

>   unique paymentID in return. The response also includes (in a true RESTful way)

Reviewer: ```suggestion
unique payment ID in return. The response also includes (in a true RESTful way)
```

>   the URIs for further actions, given the state of the payment.

Reviewer: ```suggestion
the URIs and operations for further actions, given the state of the payment.
```

>   After creating a payment, you can:

Reviewer: ```suggestion
After creating a payment, you can:
```

[mobile-pay]: /payments/mobile-pay
[seamless-view]: /payments/card/seamless-view
[redirect]: /payments/card/redirect

> [direct]: /payments/card/redirect

Reviewer: ```suggestion
```
